src/Eori/EoriValidator.php

```diff
@@ -2,21 +2,85 @@
 
 namespace CrazyFactory\Validation\Eori;
 
-use CrazyFactory\HarmonizedSystem\Countries;
+use SoapClient;
 
 class EoriValidator
 {
     /**
+     * @var SoapClient
+     */
+    private $soapClient;
+
+    /**
+     * Interpret the result as valid if match these fault codes
+     * @var array
+     */
+    private $ignoreFaultCodes = [];
+
+    /**
+     * Interpret the result as invalid if match these fault codes
+     * @var array
+     */
+    private $negativeFaultCodes = [];
+
+    public function __construct(SoapClient $client = null)
+    {
+        $this->soapClient = $client ?? new SoapClient('https://ec.europa.eu/taxation_customs/dds2/eos/validation/services/validation?wsdl');
+        $this->negativeFaultCodes = ['soap:Server'];
+        $this->ignoreFaultCodes = ['WSDL'];
+    }
+
+    /**
+     * @param array $negativeFaultCodes
+     * @return EoriValidator
+     */
+    public function setNegativeFaultCodes(array $negativeFaultCodes): EoriValidator
+    {
+        $this->negativeFaultCodes = $negativeFaultCodes;
+
+        return $this;
+    }
+
+    /**
+     * @param array $faultCodes
+     * @return EoriValidator
+     */
+    public function setIgnoreFaultCodes(array $faultCodes): EoriValidator
+    {
+        $this->ignoreFaultCodes = $faultCodes;
+
+        return $this;
+    }
+
+    /**
+     * Validate EORI against SOAP API.
+     * You can define $negativeFaultCodes and $ignoreFaultCodes to control the output that make sense for your application.
+     * If the fault code does not match any of those two settings, then exception will be thrown.
+     *
      * @param string $eori
      * @return bool
+     * @throws \SoapFault
      */
     public function validate(string $eori): bool
     {
-        $countryCode = substr($eori, 0, 2);
-        if (!Countries::isEU($countryCode) || strlen($eori) > 17 || strlen($eori) <= 2) {
-            return false;
+        try {
```

Comment:
calling api every single time in realtime is significant overhead

![image](https://user-images.githubusercontent.com/2908547/59478564-df510680-8e83-11e9-879c-54699b5e6b53.png)

so is it possible that we only call api when the old validation gives `false negative` ?

Reply:
the soft check regex for eori is `[A-Z]{2}[^\s\n\r\t]*`
if it doesnt match this format we can skip api hit and return false

Reply:
Yep we can do this since the EORI format is rarely change. (It should be very rare that they change the regex validation for EORI)